phylanx/ir/ranges.hpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2018 Parsa Amini
+// Copyright (c) 2018 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(PHYLANX_IR_RANGES)
+#define PHYLANX_IR_RANGES
+
+#include <phylanx/config.hpp>
+#include <phylanx/execution_tree/primitives/base_primitive.hpp>
+#include <phylanx/util/variant.hpp>
+
+#include <hpx/include/serialization.hpp>
+#include <hpx/include/util.hpp>
+
+#include <cstdint>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace phylanx { namespace ir
+{
+    //////////////////////////////////////////////////////////////////////////
+    class PHYLANX_EXPORT range_iterator
+      : public hpx::util::iterator_facade<range_iterator,
+            execution_tree::primitive_argument_type,
+            std::input_iterator_tag,
+            execution_tree::primitive_argument_type>
+    {
+    private:
+        using int_range_type = std::pair<std::int64_t, std::int64_t>;
+        using arg_range_type =
+            std::vector<execution_tree::primitive_argument_type>::iterator;
+        using iterator_type = util::variant<int_range_type, arg_range_type>;
+
+    public:
+        range_iterator(int_range_type it)
```

Comment:
This constructor is not used anywhere, do we need it?

Reply:
Removed.